blpapi.ts

```diff
@@ -0,0 +1,728 @@
+/// <reference path="./typings/tsd.d.ts" />
+
+'use strict';
+
+import assert = require('assert');
+import events = require('events');
+import util = require('util');
+
+import Promise = require('bluebird');
+import debug = require('debug');
+import _ = require('lodash');
+
+// Somewhat hacky but needed to import the 'blpapijs' module.
+import path = require('path');
+process.env.NODE_PATH = path.join(__dirname, 'build/Release');
+/* tslint:disable:no-var-requires */
+require('module').Module._initPaths();
+/* tslint:enable:no-var-requires */
+import blpapijs = require('blpapijs');
+
+import createError = require('custom-error-generator');
+
+// LOGGING
+var trace = debug('blpapi:trace');
+var log = debug('blpapi:debug');
+
+// PUBLIC TYPES
+
+export interface ISessionOpts extends blpapijs.ISessionOpts {
```

Comment:
Since `blpapi.ts` will eventually be compiled into js file and package with NPM, I don't think we need to export interface here anymore, it will get ripped of anyway. We can write the interface info in the `blpapi.d.ts` file in http project.